pkg/kubelet/kubeletconfig/util/files/files_test.go

```diff
@@ -0,0 +1,296 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package files
+
+import (
+	"fmt"
+	utilfs "k8s.io/kubernetes/pkg/util/filesystem"
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestFileExists(t *testing.T) {
+	fs := utilfs.NewFakeFs()
+
+	testcases := []struct {
+		name         string // name of test case
+		filePath     string // file name to be created
+		needsCreated bool   // if should create the file before test
+		isFile       bool   // is the fileName a file or folder
+		expectedBool bool   // expected result in bool
+	}{
+		{
+			name:         "not exists",
+			filePath:     "/tmp/FileExists",
+			needsCreated: false,
+			isFile:       false,
+			expectedBool: false,
+		},
+		{
+			name:         "exists as folder",
+			filePath:     "/tmp/FileExists",
+			needsCreated: true,
+			isFile:       false,
+			expectedBool: true,
+		},
+		{
+			name:         "exists as file",
+			filePath:     "/tmp/FileExists/file",
+			needsCreated: true,
+			isFile:       true,
+			expectedBool: true,
+		},
+	}
+
+	for _, testcase := range testcases {
+		var err error
+		t.Run(testcase.name, func(t *testing.T) {
+			if testcase.needsCreated {
+				if testcase.isFile {
+					// create as file
+					_, err = fs.TempFile(testcase.filePath, "")
+				} else {
+					// create as folder
+					err = fs.MkdirAll(testcase.filePath, defaultPerm)
+				}
+			}
+
+			if err != nil {
+				t.Errorf("unexpected error in creating file %s, err: %v", testcase.filePath, err)
+			} else {
+				if result, err := FileExists(fs, testcase.filePath); err != nil {
+					t.Errorf("unexpected error: %v", err)
+				} else {
+					if testcase.expectedBool != result {
+						t.Errorf("unexpected result, expected: %v, actual: %v", testcase.expectedBool, result)
+					}
+				}
+			}
+		})
+
+	}
+}
+
+func TestEnsureFile(t *testing.T) {
```

Comment:
Can you make sure at least all of the following cases are covered?
- file already exists, no action necessary
- file does not exist, EnsureFile should create the file
- file is in a subdir, neither of which exist, EnsureFile should create the subdir and file